src/main/java/upb/dice/twitter/MaxIDChecker.java

```diff
@@ -19,51 +23,59 @@
 public class MaxIDChecker {
     /**
      * This method checks for the maxID for the given keyword, Returns 1 if keyword is new
+     *
      * @param key to check the respective maxID
      * @return the maxID for the given key
      */
     public long keyParse(String key) throws IOException {
+        long result;
         Map<String, String> map = new HashMap<>();
-        try (BufferedReader br = new BufferedReader(new FileReader("Keyword_MaxID1.txt"))) {
-            String line;
-            while ((line = br.readLine()) != null) {
-                String[] splitString = line.split(",");
-                map.put(splitString[0], splitString[1]);
-            }
-            if (map.containsKey(key)) {
-                String id_string = map.get(key);
-                return new BigDecimal(id_string).intValue();
-            }
-            else {
-                return 1L;
-            }
+        File file = new File("Tweets Search Details" + File.separator + "Keyword_MaxID.txt");
+        file.getParentFile().mkdirs();
+        file.createNewFile();
+        BufferedReader br = new BufferedReader(new FileReader("Tweets Search Details" + File.separator + "Keyword_MaxID.txt"));
+        String line;
+        while ((line = br.readLine()) != null) {
+            String[] splitString = line.split(",");
+            map.put(splitString[0], splitString[1]);
         }
+        if (map.containsKey(key)) {
+            String id_string = map.get(key);
+            result = new BigDecimal(id_string).toBigInteger().longValue();
```

Comment:
Why are you using `BigDecimal`? What is the benefit in comparison to the "normal" way to parse a `long` value (i.e., using the static method of the `Long` class)?